kinoml/core/sequences.py

```diff
@@ -276,3 +280,75 @@ class RNASequence(Biosequence):
 
     ALPHABET = "AUCG"
     _ACCESSION_URL = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id={}&rettype=fasta&retmode=text"
+
+
+class KinaseDomainAminoAcidSequence(Biosequence):
+    """Biosequence for kinase domain amino acid sequences."""
+
+    ACCESSION_MAX_RETRIEVAL = None
+
+    @classmethod
+    def from_uniprot(
+        cls, *uniprot_ids: str,
+    ) -> Union[
+        "KinaseDomainAminoAcidSequence", Iterable["KinaseDomainAminoAcidSequence"], None
+    ]:
+        """
+        Retrieve kinase domain amino acid sequences of kinases defined by their Uniprot identifiers.
+        Parameters
+        ----------
+        uniprot_ids: str
+            Uniprot identifier(s). Multiple can be provided.
+        Returns
+        -------
+        kinase_domain_sequences: KinaseDomainAminoAcidSequence or list of KinaseDomainAminoAcidSequence or None
+            Retrieved kinase domain amino acid sequence(s).
+        """
+        import requests
+        import json
+
+        objects = []
+        for uniprot_id in uniprot_ids:
+            # request data
+            response = requests.get(
+                f"https://www.ebi.ac.uk/proteins/api/proteins/{uniprot_id}"
+            )
+            protein = json.loads(response.text)
+
+            # find protein kinase domains
+            for feature in protein["features"]:
+                if feature["type"] == "DOMAIN":
+                    if feature["description"] == "Protein kinase":
+                        # get kinase domain sequence details
+                        sequence = protein["sequence"]["sequence"]
+                        name = protein["id"]
+                        begin = int(feature["begin"])
+                        if begin == 1:
+                            true_N_terminus = True
+                        else:
+                            true_N_terminus = False
+                        end = int(feature["end"])
+                        if end == len(sequence):
+                            true_C_terminus = True
+                        else:
+                            true_C_terminus = False
+                        kinase_domain_sequence = sequence[begin - 1 : end]
+                        objects.append(
+                            cls(
+                                kinase_domain_sequence,
+                                name=name,
+                                metadata={
+                                    "uniprot_id": uniprot_id,
+                                    "begin": begin,
+                                    "end": end,
+                                    "true_N_terminus": true_N_terminus,
+                                    "true_C_terminus": true_C_terminus,
+                                },
+                            )
+                        )
+
+        if len(objects) == 0:
+            return None
+        elif len(objects) == 1:
+            return objects[0]
+        return objects
```

Comment:
Forgot to mention that this heterogeneous behavior in `return` types is confusing. Better conform to a single type (a list in this case), and do not surprise the user too much. This is especially useful if the conditions here are not equally frequent, because an error might look too cryptic if we assume the output is a list. For example, if no objects are returned, we return `None`, which will give a `NoneType has not __getitem__ method` error, which is veeery weird if you are just dealing with lists. Better have an `IndexError` because that's what's happening: the list is empty!

Reply:
Same applies to the case where there's a single object. Return a single item list and reduce the complexity!

Reply:
I believe this was implemented by @schallerdavid